projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/func/LLVMCallNode.java

```diff
@@ -119,19 +120,20 @@ private static int getFunctionArgumentLength(VirtualFrame frame) {
      * argument to Graal's NFI.
      *
      * @param originalArg an argument to the native function
+     * @param argType
      * @param context the context needed to retrieve a native function handle if one of the
      *            arguments is a function literal node
      * @return the conversion node that wraps the originalArg or the originalArg node itself
      * @throws LLVMUnsupportedException if one of the argument nodes is a Truffle function node that
      *             would escape to the native side
      */
-    public static LLVMExpressionNode convertToPrimitiveNode(LLVMExpressionNode originalArg, LLVMContext context) {
+    public static LLVMExpressionNode convertToPrimitiveNode(LLVMExpressionNode originalArg, LLVMType argType, LLVMContext context) {
         CompilerAsserts.neverPartOfCompilation();
-        if (originalArg instanceof LLVMAddressNode) {
```

Comment:
Let us not use nodes as an indication for types, that somehow conflicts with Truffle's idea of node rewriting.